### PR TITLE
fix(core,platform): remove css from list component, use correct class names

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
Part of #5025

#### Please provide a brief summary of this pull request.
This PR removes css from list component in core and platform.
and modifies the html template for correct use of class names.

Breaking Change:
Removed [noSeperator] property as it was not necessary, as we have [noBorder] property which has same functionality.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [NA] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/list/list.component.html [hunk 1]
```diff
@@ -4,22 +4,24 @@
...
-        <div fd-list fdInfiniteScroll class="fd-infinite-scroll" [ngStyle]="{'max-height': maxHeight}"
-            (onScrollAction)="_scrollHandler()" [scrollPercent]="scrollOffsetPercentage">
-            <ng-template *ngIf="_items.length !==0; then thenBlock else elseBlock">
-            </ng-template>
-        </div>
-    </fd-busy-indicator>
+    <ng-container *ngIf="loadMore && loadOnScroll">
+        <fd-busy-indicator size="l"
+            class="fd-busy-indicator-custom fd-list__item--growing" [block]="true" [loading]="_loading" role="alert"
+            [attr.aria-busy]="_loading" [attr.aria-labelledby]='_loading ? "loadingOnScroll" : ""'>
```
Comment (Betrozov) [inline libs/platform/src/lib/components/list/list.component.html:10]:
```suggestion
            [attr.aria-busy]="_loading" [attr.aria-labelledby]="_loading ? 'loadingOnScroll' : ''">
```

File: libs/platform/src/lib/components/list/list.component.html [hunk 2]
```diff
@@ -4,22 +4,24 @@
...
+            <div fd-list fdInfiniteScroll class="fd-list__infinite-scroll" [ngStyle]="{'max-height': maxHeight}"
```
Comment (Betrozov) [inline libs/platform/src/lib/components/list/list.component.html:11]:
```suggestion
            <div fd-list fdInfiniteScroll class="fd-list__infinite-scroll" [style.max-height]="maxHeight"
```

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/components/list/list.component.html (hunk 2 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/list/list.component.html]: fixed

File: libs/platform/src/lib/components/list/list.component.html (hunk 1 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/list/list.component.html]: fixed

**Review by Betrozov — APPROVED** [review]

File: libs/platform/src/lib/components/list/list.component.ts [hunk 3]
```diff
@@ -98,10 +98,6 @@ export class ListComponent extends CollectionBaseInput implements OnInit, AfterV
     @Input()
     noBorder = false;
 
-    /** Whether list component has removed bottom borders */
-    @Input()
-    noSeperator: boolean;
```
Comment (platon-rov) [inline libs/platform/src/lib/components/list/list.component.ts:103]: Breaking change?

Comment (DeepakSap14) [reply libs/platform/src/lib/components/list/list.component.ts:103]: Thanks. Added breaking change information in Wiki.

**Review by InnaAtanasova — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Review by puru-hk — APPROVED** [review]

**Pull request merged**